### ISSUE TITLE
Cannot set manual mipmaps for CubeTexture

### ISSUE THREAD

**Issue by valenn** [mrdoob/three.js#11671]
##### Description of the problem 

Code in "setTextureCube" seems to only handle mipmaps for compressed textures. If I replace code in path that uploads uncompressed texture with this code, it works for data textures.

```
var mipmaps = texture.mipmaps;
if (!mipmaps.length) {
	mipmaps.push(texture.image);
}
for ( var j = 0, jl = mipmaps.length; j < jl; j ++ ) {
	state.texImage2D( _gl.TEXTURE_CUBE_MAP_POSITIVE_X + i, j, glFormat, mipmaps[j][i].image.width, mipmaps[j][i].image.height, 0, glFormat, glType, mipmaps[j][i].image.data );
}
```

I haven't tested it with different types of textures. I'd be willing to make a pull request but I have no idea how this is intended to work (plain javascript without types is pretty hard to read beyond some complexity). Is there some reason why this wasn't handled? Which mipmaps should be used, those on CubeTexture or those on individual faces? (it also isn't very helpful that documentation doesn't specify what mipmaps on texture should be, ImageData or Texture? Maybe it would make things clearer)

##### Three.js version

- [x] Dev
- [ ] r85
- [ ] ...

**Comment by mrdoob** [comment]
A Pull Request would be great indeed! 👍

**Comment by mciastek** [comment]
@mrdoob I encountered similar problem. I tried to set CubeTexture's mipmaps manually, by setting `mipmaps` property of `CubeTexture` instance, but it doesn't work. There is an example how to set mipmaps manually (https://github.com/mrdoob/three.js/blob/dev/examples/webgl_materials_texture_manualmipmap.html), but it shows how to do it for normal texture, not for cubemap. Is it possible to set mipmaps manually for CubeTexture?

**Comment by robertoranon** [comment]
Sorry to bump this, but same issue here. Anybody has a working example?

**Comment by robertoranon** [comment]
[This PR](https://github.com/mrdoob/three.js/pull/15014) should do exactly what requested

**Comment by Mugen87** [comment]
Should be solved via #17072
